spec/lib/jobs/run-sku-graph-spec.js

```diff
@@ -0,0 +1,162 @@
+        var cb = job._subscribeGraphFinished.firstCall.args[0];
+
+        setImmediate(function() {
+            cb(Constants.Task.States.Succeeded);
+        });
+
+        return job._deferred
+        .then(function() {
+            expect(waterline.nodes.findByIdentifier).to.have.been.calledOnce;
+            expect(waterline.nodes.findByIdentifier).to.have.been.calledWith(fakeNode.id);
+
+            expect(waterline.skus.findOne).to.have.been.calledOnce;
+            expect(waterline.skus.findOne).to.have.been.calledWith({ id: fakeNode.sku });
+
+            expect(workflowTool.runGraph).to.have.been.calledOnce;
+            expect(workflowTool.runGraph).to.have.been.calledWith(
+                fakeNode.id,
+                fakeSku.discoveryGraphName,
+                fakeSku.discoveryGraphOptions
+            );
+
+            // Assert here that we override the sub-graphs instanceId so that
+            // our AMQP subscription to the graph finished event is actually
+            // listening on the right routing key!!!
+            expect(job.graphId).to.be.ok;
+            expect(workflowTool.runGraph.firstCall.args[2])
+                .to.have.property('instanceId')
+                .that.equals(job.graphId);
+        });
+    });
+
+    it('should fail on a failed graph', function() {
+        waterline.nodes.findByIdentifier.resolves(fakeNode);
+        waterline.skus.findOne.resolves(fakeSku);
+        var job = new RunSkuGraphJob({ nodeId: fakeNode.id }, { target: fakeNode.id }, uuid.v4());
+        job._run();
+
+        expect(job._subscribeGraphFinished).to.have.been.calledOnce;
+        var cb = job._subscribeGraphFinished.firstCall.args[0];
+
+        setImmediate(function() {
+            cb(Constants.Task.States.Failed);
+        });
+
+        return expect(job._deferred).to.be.rejectedWith(/Graph.*failed with status/);
+    });
+
+    it('should noop if there is no sku discovery graph defined', function() {
+        waterline.nodes.findByIdentifier.resolves(fakeNode);
+        waterline.skus.findOne.resolves(fakeSku);
+        var job = new RunSkuGraphJob({ nodeId: fakeNode.id }, { target: fakeNode.id }, uuid.v4());
+        job._run();
+
+        delete fakeSku.discoveryGraphName;
+        delete fakeSku.discoveryGraphOptions;
+
+        // The assertion here is that the job promise should just be resolved
+        // without having to trigger the _subscribeGraphFinished callback.
+        return job._deferred;
+    });
+
+    it('should noop if there is no sku', function() {
+        delete fakeNode.sku;
+        waterline.nodes.findByIdentifier.resolves(fakeNode);
+        var job = new RunSkuGraphJob({ nodeId: fakeNode.id }, { target: fakeNode.id }, uuid.v4());
+        job._run();
+
+        // The assertion here is that the job promise should just be resolved
+        // without having to trigger the _subscribeGraphFinished callback.
+        return job._deferred;
```

Comment:
Yeah that's what i was thinking.  Wouldn't the test throw an "unhandled rejection" if the run throws and not necessarily fail?  Maybe I'm thinking of a different scenario.